### PR TITLE
Searching Optimization

### PR DESCRIPTION
**Pull request by FilipLunski**

### CODE REVIEW

File: app/ui/SelectInput.tsx [hunk 1]
```diff
@@ -24,11 +30,35 @@ export default function SelectInput({ className }: { className?: string }) {
   // state for current search string
   const [search, setSearch] = useState<string>("");
 
+  // state for currently selected options
+  const [val, setValue] = useState<string[]>(
+    searchParams.options
+      ? [searchParams.options].flat().join(",").split(",")
```
Comment (petrvecera) [inline app/ui/SelectInput.tsx:36]: Btw isn't join and split the same thing?

File: app/ui/SelectInput.tsx [hunk 2]
```diff
@@ -66,33 +97,66 @@ export default function SelectInput({ className }: { className?: string }) {
...
-    router.replace(`${pathname}?${params.toString()}`);
+
+    // setting remDupl state
+    setRemDupl(value);
+
+    // replacing current url with new params
+    router.replace(`/?${params.toString()}`);
+
+    // prefetching result page
+    router.prefetch(`/result?${params.toString()}`);
```
Comment (petrvecera) [inline app/ui/SelectInput.tsx:145]: Interesting, so when you select just 1 PARAM we would automatically prefetch results for that?

**Review by petrvecera — APPROVED** [review]
If it work's it looks good 👍

File: app/ui/SelectInput.tsx (hunk 1 above)
Comment (FilipLunski) [reply app/ui/SelectInput.tsx:36]: As searchParams.options can be string or array of strings, we have to join potential array of strings with comma and then split it - commas are between the strings but also inside of them

File: app/ui/SelectInput.tsx (hunk 2 above)
Comment (FilipLunski) [reply app/ui/SelectInput.tsx:145]: Yes. But thanks to pointing out as now I see it prefetches even if there are no params selected, probably worth fixing

**Pull request merged**